src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts

```diff
@@ -948,21 +955,22 @@ export class SettingsEditor2 extends EditorPane {
 		return ancestors.reverse();
 	}
 
-	private updateChangedSetting(key: string, value: any, manualReset: boolean): Promise<void> {
+	private updateChangedSetting(key: string, value: any, manualReset: boolean, languageFilters: string[] | null): Promise<void> {
 		// ConfigurationService displays the error if this fails.
 		// Force a render afterwards because onDidConfigurationUpdate doesn't fire if the update doesn't result in an effective setting value change
 		const settingsTarget = this.settingsTargetsWidget.settingsTarget;
 		const resource = URI.isUri(settingsTarget) ? settingsTarget : undefined;
 		const configurationTarget = <ConfigurationTarget>(resource ? ConfigurationTarget.WORKSPACE_FOLDER : settingsTarget);
-		const overrides: IConfigurationOverrides = { resource };
+		const overrides: IConfigurationUpdateOverrides = { resource, overrideIdentifiers: languageFilters };
```

Comment:
Does this set the setting in multiple languages? It seems more likely that someone would do that by accident, especially if we only look at the first one for displaying the filtered settings.

Reply:
I changed the filters field from type `string[]` or `Set<string>` to type `string | undefined`, and changed the variable names to be singular rather than plural. This way, it could be more clear that only one language is being read for now.